packages/versioning/src/versioning.ts

```diff
@@ -58,13 +58,24 @@ export function $removed(context: DecoratorContext, t: Type, v: EnumMember) {
   }
   program.stateMap(removedOnKey).set(t, version);
 }
+
+interface RenamedFrom {
+  v: Version;
+  oldName: string;
+}
+
 export function $renamedFrom(context: DecoratorContext, t: Type, v: EnumMember, oldName: string) {
   const { program } = context;
   const version = checkIsVersion(context.program, v, context.getArgumentTarget(0)!);
   if (!version) {
     return;
   }
-  const record = { v: version, oldName: oldName };
+
+  // retrieve statemap to update or create a new one
+  const record = program.stateMap(renamedFromKey).get(t) ?? new Array<RenamedFrom>();
```

Comment:
```suggestion
  const record = getRenamedFrom(program, t) ?? [];
```

that way you don't need to do the `as`